### ISSUE TITLE
✨ Add new Face Detection Models

### ISSUE THREAD

**Issue by ishandutta0098** [ishandutta0098/mukh#30]
## Problem
As of today `mukh` supports 3 face detection models namely `blazeface`, `mediapipe` and `ultralight`. We intend to support more face detection models to make the library more comprehensive. 

## Solution
You can use this issue to suggest new models to add to `mukh` and it can be assigned to you if you are interested on working upon it.
  
Here are some criteria you can consider to select model - 
1. Select smaller and fast models
2. Recommended model should be MIT/Apache License
3. PyTorch backend
4. High Accuracy
5. No use of dlib as it causes issues in building

You can refer to existing face detection libraries and take ideas from there. It is recommend to contribute models and not use an existing package to increase the support.
  
**Design Principles of the Face Detection Module:**
Structure:
- To understand how `face_detection` module works you can refer to `mukh/face_detection`
  
```python
mukh
  |- mukh
      |- face_detection                     # Module folder
                 |- face_detection.py       # Factory Class
                 |
                 |- models                        # Folder for individual models
                      |- base_detector.py         # Base Class for Face Detection
                      |- blazeface
                      |- mediapipe
                      |- ultralight
```
  
**Inside each model folder**
- Within each of the three folders for models you will find `<model_name>_detector.py` files, the main logic for inference using that model is stored in this file. All the remaining files are taken from the actual model itself. 
- All folders and scripts within these folders should include the LICENSE information and actual source.  

## Important Note
- Please ask for the issue to be assigned before raising a PR.

## Contributing Guidelines
Refer to [contributing guide](https://ishandutta0098.github.io/mukh/contributing/) for required contribution instructions.

## Share on Socials
_If your PR regarding the issue gets merged in mukh feel free to share it on your socials and tag me, I will repost it and help increase your reach especially if you are beginner contributors!_
  
- [LinkedIn](https://www.linkedin.com/in/ishandutta0098/)
- [X](https://x.com/ishandutta0098)

**Comment by Surya-ms** [comment]
hello Ishan, I would like to work on this, can you assign this issue to me?

**Comment by 01AbhiSingh** [comment]
hii @ishandutta0098 .  I would like to contribute. assign it to me please

**Comment by KishanPipariya** [comment]
I would like to try adding RetinaFace.
Pytorch Code: https://github.com/yakhyo/retinaface-pytorch
Tensorflow Code: https://github.com/serengil/retinaface
License: MIT
Paper: [RetinaFace: Single-stage Dense Face Localisation in the Wild](https://arxiv.org/abs/1905.00641)
